containers/dotnetcore-mssql/.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,45 @@
+version: '3'
+
+services:
+  app:
+    build: 
+      context: .
+      dockerfile: Dockerfile
+      args:
+        # [Choice] Update 'VARIANT' to pick a .NET Core version: 2.1, 3.1
+        VARIANT: 3.1
+        # Options
+        INSTALL_NODE: "false"
+        NODE_VERSION: "lts/*"
+        INSTALL_AZURE_CLI: "false"
+        # On Linux, you may need to update USER_UID and USER_GID below if not your local UID is not 1000.
+        USER_UID: 1000
+        USER_GID: 1000
+
+    volumes:
+      - ..:/workspace:cached
+      
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+
+    # Uncomment the next line to use a non-root user for all processes.
+    # user: node
+
+    # Use "forwardPorts" in **devcontainer.json** to forward an app port locally. 
+    # (Adding the "ports" property to this file will not forward from a Codespace.)
+
+  db:
+    build:
+      context: ./mssql
+      dockerfile: Dockerfile
+    environment:
+      SA_PASSWORD: P@ssw0rd
+      ACCEPT_EULA: Y
+    ports:
+      - "61433:1433"
```

Comment:
This will expost port 61443 to the host, and unless you are using host networking in the other container, you won't be able to connect to it. 

There is a bug in GitHub Codespaces right now that switches the main container to host networking without modifying the database.  They're working on a fix.  Did you happen to test this there?

Reply:
You are correct, this was my workaround to get networking working in GH codespaces.  I had assumed there was a block on port 1433 for security reasons in GH codespaces and didn't consider it being a bug.
I'll update both to network_mode: host and remove the change in the port forwarding (here and in devcontainer.json)